application/src/ext/java/org/opentripplanner/ext/flex/FlexIndex.java

```diff
@@ -18,15 +20,21 @@ public class FlexIndex {
 
   private final Multimap<StopLocation, PathTransfer> transfersToStop = ArrayListMultimap.create();
 
+  private final Multimap<StopLocation, PathTransfer> transfersFromStop = ArrayListMultimap.create();
+
   private final Multimap<StopLocation, FlexTrip<?, ?>> flexTripsByStop = HashMultimap.create();
 
   private final Map<FeedScopedId, Route> routeById = new HashMap<>();
 
   private final Map<FeedScopedId, FlexTrip<?, ?>> tripById = new HashMap<>();
 
   public FlexIndex(TimetableRepository timetableRepository) {
-    for (PathTransfer transfer : timetableRepository.getAllPathTransfers()) {
+    // Flex transfers should only use WALK mode transfers.
+    StreetMode mode = StreetMode.WALK;
+    List<PathTransfer> filteredPathTransfers = timetableRepository.getAllPathTransfers().stream().filter(pathTransfer -> pathTransfer.getModes().contains(mode)).toList();
+    for (PathTransfer transfer : filteredPathTransfers) {
       transfersToStop.put(transfer.to, transfer);
+      transfersFromStop.put(transfer.from, transfer);
```

Comment:
Can you explain why this is now necessary?

Reply:
I think this is not strictly necessary, but instead it limits the available transfers to the `WALK` mode when searching for fromstops for flex transfers. Previously all of the transfers for a stop, including non-walk mode transfers, were given when using the `getTransfersFromStop` method in the `FlexRouter`. This was because it used the `findPathTransfers` method which gives all transfers for a given stop.

I think this could use my current implementation or the old one. I do have a somewhat shallow understanding of flex routing at the moment, but I went with this implementation because it seems more correct.